proxy/http/remap/PluginFactory.cc

```diff
@@ -274,13 +276,25 @@ PluginFactory::indicatePreReload()
  * @brief Tell all plugins (that so wish) that remap.config is done reloading
  */
 void
-PluginFactory::indicatePostReload(TSReturnCode reloadStatus)
+PluginFactory::indicatePostReload(bool reloadSuccessful)
 {
   Debug(_tag, "indicated config is done reloading by factory '%s' to %zu plugin%s", getUuid(), _list.count(),
         _list.count() != 1 ? "s" : "");
 
-  for (auto it = _list.begin(); _list.end() != it; it++) {
-    it->indicatePostReload(reloadStatus);
+  for (auto &plugin : _list) {
+    TSRemapReloadStatus status = TSREMAP_CONFIG_RELOAD_FAILURE;
+    if (reloadSuccessful) {
+      /* Find out which plugins (DSO) are actually instantiated by this factory */
+      std::unordered_map<PluginDso *, int> pluginUsed;
```

Comment:
This map is created for every plugin in `_list`? Can't it be promoted to outside that loop?

Reply:
fair point, thanks for catching this!